### PR TITLE
test/helpers: do not wait after killing SSH Connection

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8200)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: test/helpers/ssh_command.go [hunk 1]
```diff
@@ -269,10 +265,6 @@ func (client *SSHClient) RunCommandContext(ctx context.Context, cmd *SSHCommand)
 			if err != nil {
 				log.Errorf("write ^C error: %s", err)
 			}
-			err = session.Wait()
-			if err != nil {
-				log.Errorf("wait error: %s", err)
-			}
```
Comment (raybejjani) [inline test/helpers/ssh_command.go]: I thought the whole point of these waits was to allow non-failfast to work? if we just exit this we don't know if any state is left sitting around, no? Granted, we often wrap ssh commands in WithTimeout or WithContext but we aren't consistent... and what happens if the cleanup function times out?

**Review by raybejjani — APPROVED** [review]
We can merge this if you want, but the whole code is still the wrong way around. We can't synchronously run the command and think we can break out on a timeout reliably. If this change really does make a difference, we we can just remove the waitgroup stuff since that's where the block happens (and it is no pointless to have it).
When I tried to rewrite these functions I couldn't get my test for them to pass. Are we actually interrupting the SSH commands? or are we just returning? If we aren't interrupting then we should put a comment in there explaining why we don't wait.

**Comment by aanm** [comment]
test-me-please

**Pull request closed without merging**